### PR TITLE
ci: Add docker auth to circleci config

### LINKED ISSUE #633: [ci] Update circleci config for docker download rate limit
Truss is updating circleci configs in our repos for [docker-hub download rate limit](https://docs.docker.com/docker-hub/download-rate-limit/).  

This [post ](https://circleci.com/docs/2.0/private-images/ )shows how to make the needed change.

### PR DESCRIPTION
**Pull request by haworku**
# Summary
Update config to add auth. 

## Related Issues or PRs

closes #633 

## How To Test
- CI should run successfully

### CODE REVIEW

**Comment by haworku** [comment]
@eeeady is this it? Or am I missing something else that needs to happen here

**Review by ahobson — APPROVED** [review]
Sure looks right to me. I don't know if we need to do any config changes in CircleCi though.

**Review by pkarman — APPROVED** [review]
Confirmed that [CircleCI job](https://app.circleci.com/pipelines/github/trussworks/react-uswds/2722/workflows/576c6e56-b91d-48e9-89e3-ee01650ca337/jobs/2758) passed and that env vars are set in CI correctly.

**Comment by eeeady** [comment]
I forgot to comment before... sorry Hana!

It looks like it would work but when I check the workflow in CircleCI it doesn't seem like it's using the credentials. We might need to make changes so that the context stuff works. One moment.

**Comment by eeeady** [comment]
So if you look at the [job in circleci](https://app.circleci.com/pipelines/github/trussworks/react-uswds/2722/workflows/576c6e56-b91d-48e9-89e3-ee01650ca337/jobs/2758) it looks good but if you click into "spin up environment" you'll see the lines:

> Starting container circleci/node:12.16.2
>Warning: No authentication provided, using CircleCI credentials for pulls from Docker Hub.
> image is cached as circleci/node:12.16.2, but refreshing...
>12.16.2: Pulling from circleci/node

so the job isn't using the auth we think you set up.

I think I can make a PR against yours OR I can pull this branch down and make changes. Do you have a preference @haworku ?

**Comment by eeeady** [comment]
We might need to change branch protections to block on "build" rather than "workflow" from circleci.

**Comment by eeeady** [comment]
Made the branch protection change. Checking the build job for the output I want to see.

**Comment by eeeady** [comment]
Output is expected. no warning:

> Build-agent version 1.0.48308-8a85a016 (2020-12-10T14:02:10+0000)
> Docker Engine Version: 19.03.13
>  Kernel Version: Linux 3d864dd0b6d4 4.15.0-1077-aws #81-Ubuntu SMP Wed Jun 24 16:48:15 UTC 2020 x86_64 Linux> 
> Starting container circleci/node:12.16.2
>   image is cached as circleci/node:12.16.2, but refreshing...
> 12.16.2: Pulling from circleci/node
> Digest: sha256:7335cef9567718b3bfd03cdaaf60d457fd1ee39630d02fad3710f092e6cff072
> Status: Image is up to date for circleci/node:12.16.2
> pull stats: N/A
>  time to create container: 3.091s
>  using image circleci/node@sha256:7335cef9567718b3bfd03cdaaf60d457fd1ee39630d02fad3710f092e6cff072

**Comment by haworku** [comment]
@eeeady thanks for pointing to the place in the circleci build specifically where the warning was. Entirely missed that!

File: .circleci/config.yml [hunk 1]
```diff
@@ -29,3 +32,11 @@ jobs:
 
       - run: yarn lint
       - run: yarn danger ci --failOnErrors
+
+workflows:
+  build:
+    jobs:
+      - build:
+          context:
+          - org-global
```
Comment (pkarman) [inline .circleci/config.yml:41]: nice!

are there other repos that now need this same change?

**Review by pkarman — APPROVED** [review]
good catch @eeeady

File: .circleci/config.yml (hunk 1 above)
Comment (eeeady) [reply .circleci/config.yml:41]: Thanks for asking! I've gotten most of our other repos. I think there are a few PRs that I still need to clean up.

**Pull request merged**